src/main/java/seedu/address/model/person/Person.java

```diff
@@ -24,15 +25,19 @@ public class Person {
     private final Address address;
     private final Set<Tag> tags = new HashSet<>();
 
+    // Social media fields
+    private final Telegram telegram;
+
     /**
      * Every field must be present and not null.
      */
-    public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
+    public Person(Name name, Phone phone, Email email, Address address, Telegram telegram, Set<Tag> tags) {
```

Comment:
Just a suggestion @salty-flower.

Since we are adding multiple additional fields, modifying constructors in every PR can be a bit tricky because I add a constructor for let's say birthday and you add one for telegram, then we need to add one for both of them (which is a problem as we all are working on different features).

I would suggest adding a method such as `person.addTelegram(tele)` so that we add less breaking changes and it is much easier to handle such features.

Reply:
Yes, your point is absolutely valid. How about reducing the use of raw `Person(...)` constructor and use `PersonBuilder` instead for most of our code? It supports chaining as well which seems quite convenient.

Reply:
Yes, `PersonBuilder` would work perfectly here!

Reply:
> Yes, `PersonBuilder` would work perfectly here!

I checked again and realized `Person` is currently immutable and `PersonBuilder` is used only in test code.
Not sure if appropriate to really do that ¯\_(ツ)_/¯.

Reply:
> @salty-flower.
> 
> Since we are adding multiple additional fields, modifying constructors in every PR can be a bit tricky because I add a constructor for let's say birthday and you add one for telegram, then we need to add one for both of them (which is a problem as we all are working on different features).
> 
> I would suggest adding a method such as `person.setTelegram(tele)` so that we add less breaking changes and it is much easier to handle such features.

While it's possible to keep `Person` constructor untouched, I feel it's necessary to modify the `JsonAdaptedPerson` constructor to allow JSON serialization. Otherwise it seems that we have to implement `@JsonValue` and custom de-serialization functions, which can be messy too 🫠.

Reply:
@salty-flower Yes that's necessary as we have to specify how we want to serialize the class into an JSON object.

But that I feel is something we can do (as it's not something used by some other part of the code).